### ISSUE TITLE
Junior-1/Devops

### ISSUE THREAD

**Issue by olgaklimenko** [fullstack-development/developers-roadmap#238]
**Добавить вопросы по докеру:** 
- для чего используется (без деталей, как устроена виртуализация)
- в каком файле лежит конфиг (без деталей, как конфиг написать самому)
- как запустить с логами, как запустить в фоновом режиме, как посмотреть докер процессы, как стопнуть контейнер, как удалить контейнер.
- в чем идея docker-compose и для чего используется
- как запустить контейнеры через docker-compose, остановить, удалить.

Практика: установить докер, настроить запуск от своего юзера, запустить hello-world (стандарная штука из туториала).

**Добавить вопросы по CI:**

- В чем заключается практики CI и CD, для чего они используются, преимущества и недостатки. 

**Добавить вопросы по облачным сервисам (без уточнения провайдера):**

- что такое cloud computing platform, чем их использование отличается от поднятия собственных серверов, в чем плюсы и минусы использования.

**Comment by Znack** [comment]
Про докеры продвинутые вопросы для уровня мидла есть в риззоме, можно еще оттуда перенести :)

**Comment by Znack** [comment]
Про облачные сервисы вопросы чисто в том ключе, чем они лучше/хуже поднятия своей сетки машин?

**Comment by olgaklimenko** [comment]
Да, у меня чет нет больше идей что можно спросить без привяки к конкретному провайдеру :(

Да, для мидла из риззомы возьмем, тут ишью для 1-джуна

**Comment by Znack** [comment]
А те темы про devops, которые из нашего AWS топика, мы будем переносить, просто обобщая их? Или это просто в отдельном ишью?

Если чисто в контексте первого джуна, то может про облачные платформы вообще пока не говорить, а добавить куда-нибудь постарше и там именно пару вопросов на сравнение с обычным поднятием машин и разворачиванием всей инфраструктуры самостоятельно :)

**Comment by stanislav-az** [comment]
Для джуна(1,2,3) думаю вообще не нужны вопросы по облачным платформам, так как нужна практика использования.
Я бы оставил вопросы про docker-compose для jun-1, а все остальное про докер на jun-2.

**Comment by olgaklimenko** [comment]
В чате мы предлагали AWS вообще убрать из уровней и сделать просто доп знаниями и пришли к тому что надо просто заменить вопросы на вопросы без привязки к провайдеру. Чего это я решила их в 1го джуна положить, запуталась немного :)

**Comment by olgaklimenko** [comment]
По докерам, я думаю, все вопросы нужны из предложеных на 1м джуне, потому что докер мы используем на проекте. И есть смысл либо добавлять все, либо не добавлять ничего и просто заставлять читать readme всех кто приходит на проект и знать 2 команды как запустить и как завершить.

**Comment by olgaklimenko** [comment]
Вообще понимание тем по девопсу зависят сильно от того насколько ты их используешь. Именно потому я предлагала вынести их из карты развития и добавлять опционально когда они нужны. Типа пошел на новый проект где это нужно, открыл доп карту развития и изучил, потом пошел работать с этими штуками.

**Comment by kelizarov** [comment]
Я предлагаю 1 джуна не трогать вообще, и начинать спрашивать со 2-ого джуна. Мы же в нем уже и линукс спрашиваем, так что там вопросы по докеру будут как раз. В остальном претензии не имею.

**Comment by olegromashin** [comment]
Согласен с тем, что на первом джуне это не надо. Например мои текущие знания можно уложить в два-три небольших абзаца и я от этого вообще не страдаю. Предлагаю сделать как выше написала @olgaklimenko - написать краткое руководство и отправлять читать при необходимости.

**Comment by Gmihtt** [comment]
Внесу свои пару слов, может тогда сделать это как свободную тему, но все же указать, что человеку который сдает 2 джуна нужно это знать?
Типо без конкретных требований, что отвечать и на какие вопросы, а в целом чтоб джун мог объяснить зачем это, как это и для чего это.
Мне кажется своими словами это даже лучше, чем конкретные вопросы
Ну и оставить формулировки Оли + какие-то ссылки как подсказки с чего начать и куда копать

**Comment by kelizarov** [comment]
Мне кажется делать свободные темы на джуне не стоит. Во-первых сдающему будет сложно определить насколько глубоко ему тогда надо будет копаться в теме чтобы успешно сдать или наоборот поверхостно пройдет по ним. Во-вторых принимающим будет сложно определить насколько сдающий разбирается в теме. Составляя вопросы хотя бы поставить обоих в конкретные рамки и будет проще работать с процессом сдачи, и заодно укрепит знания необходимые.

**Comment by Znack** [comment]
Вообще может нам и правда просто забить на девопс именно в нашей карте, но поискать в интернете? мне кажется, должны быть роадмапы по тем же докеру и облачным платформам. Если где-то на проекте это будет вылазить или кому-то лично будет интересно, то просто будем отправлять в эти роадмапы и всё :)

**Comment by stanislav-az** [comment]
Нашел классный курс по докеру, надо бы добавить в источники, когда раздел будет готов: [learndocker.online](https://learndocker.online)
